lib/ansible/modules/cloud/alicloud/alicloud_instance.py

```diff
@@ -0,0 +1,680 @@
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: alicloud_instance
+version_added: "2.5"
+short_description: Create, Start, Stop, Restart or Terminate an Instance in ECS. Add or Remove Instance to/from a Security Group.
+description:
+    - Create, start, stop, restart, modify or terminate ecs instances.
+    - Add or remove ecs instances to/from security group.
+options:
+    state:
+      description:
+        - The state of the instance after operating.
+      default: 'present'
+      choices: [ 'present', 'running', 'stopped', 'restarted', 'absent' ]
+    alicloud_zone:
+      description:
+        - Aliyun availability zone ID in which to launch the instance.
+          If it is not specified, it will be allocated by system automatically.
+      aliases: ['acs_zone', 'ecs_zone', 'zone_id', 'zone' ]
+    image_id:
+      description:
+        - Image ID used to launch instances. Required when C(state=present) and creating new ECS instances.
+      aliases: [ 'image' ]
+    instance_type:
+      description:
+        - Instance type used to launch instances. Required when C(state=present) and creating new ECS instances.
+      aliases: [ 'type' ]
+    group_id:
+      description:
+        - Security Group id used to launch instance or join/remove existing instances to/from the specified Security Group.
+      aliases: [ 'security_group_id' ]
+    vswitch_id:
+      description:
+        - The subnet ID in which to launch the instances (VPC).
+      aliases: ['subnet_id']
+    instance_name:
+      description:
+        - The name of ECS instance, which is a string of 2 to 128 Chinese or English characters. It must begin with an
+          uppercase/lowercase letter or a Chinese character and can contain numerals, ".", "_" or "-".
+          It cannot begin with http:// or https://.
+      aliases: ['name']
+    description:
+      description:
+        - The description of ECS instance, which is a string of 2 to 256 characters. It cannot begin with http:// or https://.
+    internet_charge_type:
+      description:
+        - Internet charge type of ECS instance.
+      default: "PayByBandwidth"
+      choices: ["PayByBandwidth", "PayByTraffic"]
+    max_bandwidth_in:
+      description:
+        - Maximum incoming bandwidth from the public network, measured in Mbps (Mega bit per second).
+      default: 200
+    max_bandwidth_out:
+      description:
+        - Maximum outgoing bandwidth to the public network, measured in Mbps (Mega bit per second).
+      default: 0
+    host_name:
+      description:
+        - Instance host name.
+    password:
+      description:
+        - The password to login instance. After rebooting instances, the modified password would be take effect.
+    disk_category:
+      description:
+        - Category of the system disk.
```

Comment:
Are there any disks apart from system disks?

If so then this should be named `system_disk_category` and remove the aliases.